notes/stv/20230125-red-deploy-01.txt

```diff
@@ -0,0 +1,952 @@
+	    },
+	    {
+	      "id": "2HQGZ3B2V",
+	      "path": "/Users/chee2ooW5rieSohv/examples/7. Good astrometric solutions via ML Random Forest classifier"
+	    },
+	    {
+	      "id": "2HQNGTM42",
+	      "path": "/Users/chee2ooW5rieSohv/examples/8. Tips and tricks"
+	    }
+	  ]
+	}
+
+
+
+
+
+# -----------------------------------------------------
+# Run the ML RandomForest classifier example.
+#[root@ansibler]
+
+    source /deployments/zeppelin/bin/zeppelin-rest-tools.sh
+
+    testall \
+        "${username}" \
+        "${password}" \
+        "/Users/${username}/examples/7" \
+    | tee "/tmp/${username}-testone.json" \
+    | jq '
+       .notebooks[] | {
+            "id": .noteid,
+            "name": .execute.name,
+            "path": .execute.path,
+            "duration": .execute.duration,
+            "paragraphs": [
+                .execute.paragraphs[] | {
+                    "title": .title,
+                    "duration": .duration
+                    }
+                ]
+            }
+       '
+       
+       # Hangs..
+      
+       # Run test from UI
+       # Success
+        
+        > 
+             "title": "Raw catalogue with selected columns",
+             "duration": "0:4:4"
+             
+              "title": "Train up the Random Forest",
+             "duration": "0:2:39"
+             
+
+   
+# -----------------------------------------------------
+# Create our benchmark script.
+#[root@ansibler]
+
+cat > /tmp/run-benchmark.py << 'EOF'
+#!/bin/python3
+import sys
+from aglais_benchmark import AglaisBenchmarker
+
+try:
+
+    opts = [opt for opt in sys.argv[1:] if opt.startswith("-")]
+    args = [arg for arg in sys.argv[1:] if not arg.startswith("-")]
+
+    endpoint = args[0]
+    testconfig = args[1]
+    userlist = args[2]
+    usercount = int(args[3])
+    delaystart = int(args[4])
+    delaynotebook = int(args[5])
+
+except IndexError:
+
+    raise SystemExit(f"Usage: {sys.argv[0]} <Zepelin endpoint> <test config> <list of users> <number of users>")
+
+print("{")
+print(
+"""
+\"config\": {{
+    \"endpoint\":   \"{}\",
+    \"testconfig\": \"{}\",
+    \"userlist\":   \"{}\",
+    \"usercount\":  \"{}\",
+    \"delaystart\":  \"{}\",
+    \"delaynotebook\":  \"{}\"
+    }},
+\"output\":
+""".format(
+        endpoint,
+        testconfig,
+        userlist,
+        usercount,
+        delaystart,
+        delaynotebook
+        )
+    )
+
+print("---start---")
+AglaisBenchmarker(
+    testconfig,
+    userlist,
+    "/tmp/",
+    endpoint
+    ).run(
+        concurrent=True,
+        users=usercount,
+        delay_start=delaystart,
+        delay_notebook=delaynotebook
+        )
+print("---end---")
+print("}")
+EOF
+
+chmod 'a+x' /tmp/run-benchmark.py            
+
+
+# -----------------------------------------------------
+# Create test users
+#[root@ansibler]
+             
+source /deployments/zeppelin/bin/create-user-tools.sh
+
+import-test-users
```

Comment:
That test was run on live, but what I'm pretty sure happened was that the test was run as "chee2ooW5rieSohv". Reason for the mixup was that at first I created a benchmarker user config for chee2ooW5rieSohv, but that failed because the benchmarker expected a the test user configuration that is generated in /tmp when running "import-test-users". I then ran the import-test-users and re-ran the test, but probably due to the Manila share issue the user wasn't created, but I didn't notice as the test was run, but was run as the original test user. Complicated & should have included the notes on the initial failed test.. Also need to modify the benchmarker to decouple it from the dependence on the test user config.

Reply:
Actually the Reyesfan user was created, as they have an entry in the shiro database as well as a notebook. Its the share & home directory that wasn't created. So I think the test could have still run as them I think

Reply:
OK, makes sense, I think.
I've added some gardening issues to avoid this happening again.
* #1111 
* #1110